packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewManagerImpl.kt

```diff
@@ -94,16 +95,31 @@ internal object LottieAnimationViewManagerImpl {
                         view.reverseAnimationSpeed()
                     }
                 }
+            } else {
+                val actualStartFrame = view.composition?.startFrame?.toInt()
+                val actualEndFrame = view.composition?.endFrame?.toInt()
+
+                val minFrame = view.minFrame.toInt()
+                val maxFrame = view.maxFrame.toInt()
+                if (actualStartFrame != null && actualEndFrame != null && (minFrame != actualStartFrame || maxFrame != actualEndFrame)) {
+                    view.setMinAndMaxFrame(actualStartFrame, actualEndFrame)
+                }
```

Comment:
Can you explain what is happening here?

Reply:
This makes sure that if a `play` methods is called without start/end, it really plays it from the original asset's start till its end. If this was not done and the user calls `play(<start-frame>, <end-frame>)` then wants to play the entire thing with `play`, then this second play would not play from the very beg till the very end. Instead it would play from the prev passed `play` call because of the `setMinAndMaxFrame`.

Here simply we are making sure the min and max is reset to the original if no start/end is passed.

Reply:
Thank you so much! However, I think we need to refactor this code because it can lead to race conditions. While this piece serves as a workaround for now, the root issue, in my opinion, is the race condition. Especially, on line 85, we are scheduling a task on the main thread, and calling `play` multiple times can result in conflicting configurations on the view.